### ISSUE TITLE
Canvas, bitmap rendering context and readback

### ISSUE THREAD

**Issue by svgeesus** [w3c/ColorWeb-CG#5]
Originally raised on the [TAG review of Lab](https://github.com/w3ctag/design-reviews/issues/488#issuecomment-610107610)

@dbaron wrote:

> ImageBitmapRenderingContext
> including readback APIs (e.g., the format of raw bitmap data that you get back)

discussion of those issues to be added to [Canvas API](https://w3c.github.io/ColorWeb-CG/#canvas-api) seciton of the report

**Comment by svgeesus** [comment]
originally [noted here](https://github.com/w3ctag/design-reviews/issues/488#issuecomment-618105500)

> [gs to audit](https://github.com/w3ctag/design-reviews/issues/488#issuecomment-610107610) to ensure that the platform has good support throughout for out-of-sRGB colors:
> 
>     * (this one I'm pretty sure about) everything to do with [canvas pixel manipulation](https://html.spec.whatwg.org/multipage/#pixel-manipulation), including the [`ImageData` interface](https://html.spec.whatwg.org/multipage/#imagedata), to have variants that allow getting canvas data that's outside of the sRGB gamut
> 
>     * the [algorithm for serializing bitmaps](https://html.spec.whatwg.org/multipage/#serialising-bitmaps-to-a-file), to ensure that it's possible to serialize any possible canvas accurately
> 
>     * the [`<feColorMatrix>`](https://drafts.fxtf.org/filter-effects/#feColorMatrixElement) and [`<feComponentTransfer>`](https://drafts.fxtf.org/filter-effects/#feComponentTransferElement) filter primitives, to ensure that there are sensible ways to do those operations on colors outside of the sRGB gamut

**Comment by palemieux** [comment]
@ccameron-chromium Does the proposal at https://github.com/WICG/canvas-color-space/blob/master/CanvasColorSpaceProposal.md include a readback API as suggested by TAG?

**Comment by ccameron-chromium** [comment]
Yes, that adds a mechanism for reading back data outside of the sRGB gamut (ImageData can be uint8, uint16, or float32 now, and its gamut may be srgb, display-p3, or rec2020). Of note is that float32 ImageData can use numbers outside of the range of [0, 1] to represent any color (in any gamut).

The spec also says "The functions toDataURL and toBlob should produce resources that best match the fidelity of the underlying canvas (subject to the limitations of the implementation and the requested format)."
